altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -253,12 +254,54 @@ mod tests {
   #[test]
   fn only_critical_task() {
     let _g = test::set_up();
-    let handle_1 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 1");
-    let handle_2 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 2");
+    test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
+    test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
     start_scheduler();
-    for i in 0..100 {
-      assert!(test::current_task().unwrap().priority == Priority::Critical);
+    for _i in 0..100 {
+      assert!(test::current_task().unwrap().priority == Priority::Normal);
     }
   }
 
+  #[test]
+  //No task, but scheduler is run
+  fn no_task_sched_run_sched() {
+    let _g = test::set_up();
+    start_scheduler();
+    assert_eq!(test::current_task().unwrap().priority, Priority::__Idle);
```

Comment:
Instead of looking at the priority of the task you may want to just check that `current_task().is_some()`, unless you're explicitly checking to see if the idle task is scheduled?